docs/why.qmd

```diff
@@ -2,24 +2,313 @@
 title: "Why Ibis?"
 ---
 
+Ibis defines a Python dataframe API that supports any query engine -- the
+frontend for any backend data platform. This includes 20+ backends today, most
+SQL-based.  This allows Ibis to have excellent performance -- as good as the
+backend it is connected to -- with a consistent user experience.
+
+## What is Ibis?
+
 Ibis is the portable Python dataframe library.
 
-If you've had issues with scaling data transformation code in Python, need to
-work with data in multiple data platforms, find yourself translating between
-other Python dataframe APIs, or just want a great Python dataframe experience,
-Ibis is for you.
+We can demonstrate this with a simple example on a few local backends:
+
+```{python}
+import ibis
+
+ibis.options.interactive = True
+```
+
+::: {.panel-tabset}
+
+```{python}
+#| echo: false
+t = ibis.examples.penguins.fetch()
+t.to_parquet("penguins.parquet")
+```
+
+## DuckDB
+
+```{python}
+con = ibis.duckdb.connect()  # <1>
+
+t = con.read_parquet("penguins.parquet")
+t.limit(3)
+```
+
+1. Change only your connection to switch between backends.
+
+```{python}
+t.group_by(["species", "island"]).agg(count=t.count()).order_by("count")
+```
+
+## Polars
+
+```{python}
+con = ibis.polars.connect()  # <1>
+
+t = con.read_parquet("penguins.parquet")
+t.limit(3)
+```
+
+1. Change only your connection to switch between backends.
+
+```{python}
+t.group_by(["species", "island"]).agg(count=t.count()).order_by("count")
+```
+
+## DataFusion
+
+```{python}
+con = ibis.datafusion.connect()  # <1>
+
+t = con.read_parquet("penguins.parquet")
+t.limit(3)
+```
+
+1. Change only your connection to switch between backends.
+
+```{python}
+t.group_by(["species", "island"]).agg(count=t.count()).order_by("count")
+```
+
+## PySpark
+
+```{python}
+con = ibis.connect("pyspark://")  # <1>
+
+t = con.read_parquet("penguins.parquet")
+t.limit(3)
+```
+
+1. Change only your connection to switch between backends.
+
+```{python}
+t.group_by(["species", "island"]).agg(count=t.count()).order_by("count")
+```
+
+:::
+
+## Who is Ibis for?
+
+Ibis for data engineers, data analysts, and data scientists to use directly with
+their data platform(s) of choice. It also has benefits for [data
+platforms](#ibis-for-data-platforms),
+[data organizations](#ibis-for-data-organizations), and
+[data library developers](#ibis-for-data-library-developers).
+
+### Ibis for data practitioners
+
+You can use Ibis at any stage of your data workflow, no matter your role.
+
+**Data engineers** can use Ibis to:
+
+- write and maintain complex ETL/ELT jobs
+- replace fragile SQL string pipelines with a robust Python API
+- replace PySpark with a more Pythonic API that supports Spark and many other
+ backends
+
+**Data analysts** can use Ibis to:
+
+- use Ibis interactive mode for rapid exploration
```

Comment:
maybe we can reword this?  it currently reads as "... can use Ibis to use Ibis interactive mode... "

how about
```suggestion
- perform rapid exploratory data analysis using interactive mode
```